gems/pending/spec/appliance_console/database_replication_primary_spec.rb

```diff
@@ -0,0 +1,111 @@
+      expect(subject).to_not receive(:confirm_reconfiguration)
+      expect(subject).to receive(:confirm).and_return(true)
+      expect(subject.ask_questions).to be true
+    end
+
+    it "returns true when confirm_reconfigure and input is confirmed" do
+      expect(subject).to receive(:ask_for_unique_cluster_node_number)
+      expect(subject).to receive(:ask_for_database_credentials)
+      expect(subject).to receive(:repmgr_configured?).and_return(true)
+      expect(subject).to receive(:confirm_reconfiguration).and_return(true)
+      expect(subject).to receive(:confirm).and_return(true)
+      expect(subject.ask_questions).to be true
+    end
+
+    it "returns false when confirm_reconfigure is canceled" do
+      expect(subject).to receive(:ask_for_unique_cluster_node_number)
+      expect(subject).to receive(:ask_for_database_credentials)
+      expect(subject).to receive(:repmgr_configured?).and_return(true)
+      expect(subject).to receive(:confirm_reconfiguration).and_return(false)
+      expect(subject).to_not receive(:confirm)
+      expect(subject.ask_questions).to be false
+    end
+
+    it "returns false when input is not confirmed" do
+      expect(subject).to receive(:ask_for_unique_cluster_node_number)
+      expect(subject).to receive(:ask_for_database_credentials)
+      expect(subject).to receive(:repmgr_configured?).and_return(false)
+      expect(subject).to_not receive(:confirm_reconfiguration)
+      expect(subject).to receive(:confirm).and_return(false)
+      expect(subject.ask_questions).to be false
+    end
+  end
+
+  context "#activate" do
+    it "returns true when configure succeed" do
+      expect(subject).to receive(:generate_cluster_name).and_return(true)
+      expect(subject).to receive(:create_config_file).and_return(true)
+      expect(subject).to receive(:initialize_primary_server).and_return(true)
+      expect(subject.activate).to be true
+    end
+
+    it "returns false when create_config_file fails" do
+      expect(subject).to receive(:generate_cluster_name).and_return(true)
+      expect(subject).to receive(:create_config_file).and_return(false)
+      expect(subject).to_not receive(:initialize_primary_server)
+      expect(subject.activate).to be false
+    end
+
+    it "returns false when generate_cluster_name fails" do
+      expect(subject).to receive(:generate_cluster_name).and_return(false)
+      expect(subject).to_not receive(:create_config_file)
+      expect(subject).to_not receive(:initialize_primary_server)
+      expect(subject.activate).to be false
+    end
+
+    it "returns false when initialize_primary_server fails" do
+      expect(subject).to receive(:generate_cluster_name).and_return(true)
+      expect(subject).to receive(:create_config_file).and_return(true)
+      expect(subject).to receive(:initialize_primary_server).and_return(false)
+      expect(subject.activate).to be false
+    end
+  end
+
+  context "#initialize_primary_server" do
```

Comment:
Actually it might be causing the test failure on `miq_generic_mount_session` because it is checking `$?` which might be an error if that test runs after the "raises an error when REGISTER_CMD fails" spec here.